kappa/event_source/s3.py

```diff
@@ -28,7 +28,10 @@ def __init__(self, context, config):
         self._lambda = kappa.awsclient.create_client('lambda', context.session)
 
     def _make_notification_id(self, function_name):
-        return 'Kappa-%s-notification' % function_name
+        id_no = self._config.get('id')
```

Comment:
This could instead use the Arn since it's already a uuid, no?

So in `_get_notification_spec`,
```
notification_spec = {
    'Id': self._make_notification_id(function.arn),
    'Events': [e for e in self._config['events']],
    'LambdaFunctionArn': function.arn,
}
```

Reply:
The alternative, what you have, requires anyone using kappa to provide an Id.

Reply:
For context - I have a PR open to upgrade this library in Zappa.
https://github.com/Miserlou/Zappa/pull/1518